### ISSUE TITLE
What is the purpose of the `ExforBaseParser` class?

### ISSUE THREAD

**Issue by DerPhysikeR** [IAEA-NDS/exfor-parserpy#4]
Hi

`ExforBaseParser` doesn't keep track of any state (it doesn't even have an `__init__` method) therefore it seems to me like the interface to the parser could be simplified by just having a `read_exfor` function for reading exfor files instead of a class which has to be instantiated first.

This would reduce the overhead for reading exfor files from

```python
parser = ExforBaseParser()
exfor_entry = parser.readfile(TEST_DATA / 'entry_21308.txt')
```

to

```python
exfor_entry = read_exfor(TEST_DATA / 'entry_21308.txt')
```

For writing exfor files a function like `write_exfor(file_to_write_to, exfor_entry)` could make sense, although there are two things I would look at first:

 - Is any dictionary valid for writing it to an exfor file?
 If not, it might be better to have a custom `Exfor` class, which can be created from a dict which is validated on instantiation.
 By implementing methods like `__getitem__` (and possibly `__setitem__`, although I prefer immutable objects) it could behave essentially the same as a dictionary.
 In this case it might be better to have a `write_exfor` method of this custom class, because it minimizes the number of function arguments.
 This would be similar to pandas, which has a `pd.read_csv` function to construct a `DataFrame` object from a file, but the `DataFrame` object itself has a `to_csv` method for writing `.csv` files.

 - Should file reading/writing be handled by the parser at all?
 I would point to Pythons handling of json data as an example: https://docs.python.org/3/library/json.html
 The json module has methods for loading json data from a string `json.loads` and for dumping json data to a string `json.dumps`, alternatively it can directly interact with file streams using `json.load` and `json.dump`, but it is never actually responsible for opening and closing files directly.

Also, I would get rid of the `overwrite` flag completely, to reduce the packages surface area even further, since opening a file and writing to it in any other case just overwrites it by default.
This is something the user should handle beforehand if they want to have different behavior.

**Comment by gschnabel** [comment]
Hello

Thank you for these suggestions!

About the class: I started with a class because I assumed that some state would be required at some point but it didn't happen. I agree that it can be removed and simplifies the use of the package.

About writing EXFOR files: Not all dictionaries would translate to valid EXFOR files. In fact, many constraints exists and it is quite a task to comply with all of them when creating a dictionary from scratch. For instance, the EXFOR formats manual describes some constraints but there are also so-called EXFOR dictionaries that contain rules for things, such as which physical units are allowed and what strings to use to represent specific aspects of an experiment, e.g., (TOF) for time-of-flight. The writing functionality was mainly implemented to allow for rigorous testing of the translation without knowing too much about the format (see tests/test_bijection.py). In the future, the writing functionality may be useful to some users (in particular, people that read papers and reports of experiments and produce such files are called *compilers* in the nuclear data community) but its further development beyond what is available now is not a priority for me at the moment. However, if validation of a dictionary becomes a requirement, the approaches you are suggesting sound very pertinent to me.

About whether reading or writing should be handled by the parser: The json package is well known and if we follow its philosophy with regard to reading and writing files, it may indeed help the user. Then one can also get rid of the overwrite argument.

Would you like to implement the changes suggested by you? The scripts in the `examples/` and `tests/` folder would need to be updated accordingly as well.

**Comment by DerPhysikeR** [comment]
Sounds good to me, I will take care of the implementation.

For simplicity's sake, I will probably create a `read_exfor` function for reading a file, a `write_exfor` function for directly writing a file and a `to_exfor` function for dumping an exfor compatible dictionary into a string analogous to `json.dumps`.
This should cover most use cases.

**Comment by gschnabel** [comment]
Successfully merged your pull request #5 containing your modifications. Thank you!
